### PR TITLE
chore: move firecracker bits to dedicated library

### PR DESCRIPTION
**Pull request by sprutton1**
This creates an `si-firecracker` library and moves all of the `local_uds`-specific work into it. None of the behavior is different, I'm just moving things around. I have a long-term goal to start moving the bash bits into rust, so this will make that change easier later.

### CODE REVIEW

**Comment by sprutton1** [comment]
bors try

**Comment by sprutton1** [comment]
bors merge

File: dev/Tiltfile [hunk 1]
```diff
@@ -232,11 +232,11 @@ local_resource(
     "veritech",
     labels = ["backend"],
     cmd = cmd,
-    serve_cmd = serve_cmd,
+    # serve_cmd = serve_cmd,
     serve_env = {"SI_FORCE_COLOR": "true"},
     # This is the serve command you might need if you want to execute on firecracker for 10 functione executions.
     # NB: BUCK2 MUST RUN AS ROOT OR THIS WILL NOT WORK
-    # serve_cmd = "SI_LOG=debug buck2 run {} -- --cyclone-local-firecracker --cyclone-pool-size 10".format(veritech_target),
+    serve_cmd = "SI_LOG=debug buck2 run {} -- --cyclone-local-firecracker --cyclone-pool-size 10".format(veritech_target),
```
Comment (nickgerace) [inline dev/Tiltfile:239]: Will break macOS users?

**Comment by sprutton1** [comment]
bors cancel

**Comment by sprutton1** [comment]
bors merge

**Pull request merged**